src/Controller/SiteInfoController.php

```diff
@@ -0,0 +1,203 @@
+          return $collection;
+        }
+
+        if (isset($field_definitions[$field_name]) && $field_definitions[$field_name]->isRequired()) {
+          $collection[$field_name] = $field_info;
+        }
+
+        return $collection;
+      },
+      []
+    );
+
+    return $required_fields;
+  }
+
+  /**
+   * Get bundles ordered by number of instances.
+   *
+   * @param array $data
+   *   The collected data from sampler module for entity type.
+   *
+   * @return array
+   *   Returns bundle names ordered by count.
+   */
+  protected function getBundlesByCount(array $data) {
+    $bundles_by_count = [];
+
+    foreach ($data as $bundle_name => $bundle_info) {
+      if (!isset($bundle_info['instances'])) {
+        continue;
+      }
+
+      $bundles_by_count[$bundle_name] = $bundle_info['instances'];
+    }
+
+    arsort($bundles_by_count);
+
+    return $bundles_by_count;
+  }
+
+  /**
+   * Get bundles ordered by number of instances.
+   *
+   * @param array $data
+   *   The collected data from sampler module for entity type.
+   *
+   * @return array
+   *   Returns bundle names ordered by count.
+   */
+  protected function getBundlesByNumberOfFields(array $data) {
+    $bundles_by_number_of_fields = [];
+
+    foreach ($data as $bundle_name => $bundle_info) {
+      $bundles_by_number_of_fields[$bundle_name] = count(array_keys($bundle_info['fields']));
+    }
+    arsort($bundles_by_number_of_fields);
+
+    return $bundles_by_number_of_fields;
+  }
+
+  /**
+   * Gets site information.
+   *
+   * Allowed query options are following:
+   * => rule: {string} - options:
+   *   - count: to sort bundles by number of instance
+   *   - number_of_fields: to sort bundles by number of fields
+   * Default: "count"
+   *
+   * => index: {int}
+   * Default: 0
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The symfony request object.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   Returns JSON with site information needed by performance tests.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function siteInfo(Request $request) {
+    $rule = $request->query->get('rule', 'count');
+    $index = (int) $request->query->get('index', '0');
+
+    $data = (array) $this->cache()
+      ->get('thunder-performance-measurement:site-info:node');
+    if (!isset($data['data'])) {
+      $this->samplerMapping->enableMapping(FALSE);
+      $data = $this->samplerPluginManager->createInstance('bundle:node')->collect();
+      $data['bundles_by_count'] = $this->getBundlesByCount($data);
+      $data['bundles_by_number_of_fields'] = $this->getBundlesByNumberOfFields($data);
+
+      $this->cache()
+        ->set('thunder-performance-measurement:site-info:node', $data);
+    }
+    else {
+      $data = $data['data'];
+    }
+
+    $top_bundle = array_keys($data['bundles_by_count'])[$index];
```

Comment:
This is not necessary to do if `$rule === 'bundle_of_fields'`, I would put it into an else, of the if statement below.